File: Problems/2405. (M) Optimal Partition of String/Python3/solution.py
class Solution:
  def partitionString(self, s: str) -> int:
    # greedy

    cur = set()
    ret = 0
    for c in s:
      if c not in cur:
        cur.add(c)
      else:
        ret += 1
        cur.clear()
        cur.add(c)
    
    return ret if not cur else ret + 1

File: Problems/2405. (M) Optimal Partition of String/Java/solution.java
class Solution {
  public int partitionString(String s) {
    // greedy

    Set<Character> cur = new HashSet();
    int ret = 0;
    for (int i = 0; i < s.length(); i++) {
      if (!cur.contains(s.charAt(i)))
        cur.add(s.charAt(i));
      else {
        ret++;
        cur.clear();
        cur.add(s.charAt(i));
      }
    }
    return cur.size() > 0 ? ++ret : ret;
  }
}
